src/sage/calculus/functions.py

```diff
@@ -147,4 +147,16 @@ def jacobian(functions, variables):
     if not isinstance(variables, (tuple, list, Vector)):
         variables = [variables]
 
-    return matrix([[diff(f, v) for v in variables] for f in functions])
+    jacobian_matrix([[diff(f, v) for v in variables] for f in functions])
```

Comment:
I think you meant:
```suggestion
    jacobian_matrix = matrix([[diff(f, v) for v in variables] for f in functions])
```

Reply:
But with the other changes I suggested you should just revert the `jacobian` function back to how it was, and add a `hessian` function for that functionality.